core/src/constants/mod.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! Types associated with accessing constants.
+
+mod constant_address;
+
+pub use constant_address::{dynamic, Address, ConstantAddress, DynamicAddress};
+
+use alloc::borrow::ToOwned;
+
+use crate::{
+    metadata::{DecodeWithMetadata, MetadataExt},
+    Error, Metadata, MetadataError,
+};
+
+/// Run validation logic against some constant address you'd like to access. Returns `Ok(())`
+/// if the address is valid (or if it's not possible to check since the address has no validation hash).
+/// Return an error if the address was not valid or something went wrong trying to validate it (ie
+/// the pallet or constant in question do not exist at all).
+pub fn validate_constant<Address: ConstantAddress>(
```

Comment:
Use in `subxt::constants::COnstantsClient::validate()`?

Reply:
yes, perhaps it's better to move this to the constant client stuff